packages/react-bindings/src/contexts/StreamVideoContext.tsx

```diff
@@ -30,6 +36,26 @@ export const StreamVideo = ({
   language,
   translationsOverrides,
 }: PropsWithChildren<StreamVideoProps>) => {
+  const prevClient = useRef<StreamVideoClient | undefined>(undefined);
+
+  useEffect(() => {
+    if (prevClient.current) {
+      prevClient.current
+        .disconnectUser()
+        .catch((error) => console.error(`Failed to disconnect`, error));
+    }
+
+    if (client.user) {
+      client
```

Comment:
Just a question: Shouldn't we be waiting until disconnect happens? In the past, we had some problems in chat with quick successive connects and disconnects.